src/base.ts

```diff
@@ -624,7 +752,16 @@ export default class KintoClientBase {
    *     when faced with transient errors.
    * @return {Promise<Object, Error>}
    */
-  async createBucket(id, options = {}) {
+  async createBucket(
+    id: string | null,
+    options: {
+      data?: any;
```

Comment:
Shouldn't it be `Object` here? (ie. it can't be a string or a number...)

Reply:
I introduced a new type, `MappableObject` that will reject if someone passes in an array. Unfortunately TypeScript doesn't have a built-in "object, but not array" type.